pkg/terraform/provider_installer.go

```diff
@@ -38,6 +38,27 @@ func (p *ProviderInstaller) Install() (string, error) {
 	providerDir := p.getProviderDirectory()
 	providerPath := p.getBinaryPath()
 
+	_, err := os.Stat(providerDir)
+	if os.IsNotExist(err) {
+		logrus.WithFields(logrus.Fields{
+			"path": providerDir,
+		}).Debug("Provider directory destination not found, creating ...")
+		if err := os.MkdirAll(providerDir, 0755); err != nil {
+			return "", err
```

Comment:
I wonder if we should add some context to the error when we fail to create or write the directory ?